src/views/about/index.js

```diff
@@ -1,12 +1,22 @@
 import { NavBar } from '../../components'
+import React, { useContext } from 'react'
+import { AuthContext } from '../../providers/use-auth'
+import { useNavigate } from 'react-router'
 
 export const About = () => {
-  return (
-    <div>
-      <NavBar />
-      <h4>
-        About...!
-      </h4>
-    </div>
-  )
+  const { user } = useContext(AuthContext)
```

Comment:
Prefer this pattern...!

```jsx
import { Navigate } from "react-router-dom";

// This is a *guard return*, it deals with null data
if (!user) return <Navigate to="/login" />
// Notice I use <Navigate /> - This make our function return only one type of data: JSX

// Just return normally
return (
  // About JSX here
)
```

Reply:
I resolved, but still dont know why it works

Reply:
> It's pretty well explained in this browser warning...!

![image](https://user-images.githubusercontent.com/39024711/180970897-a743f640-6b96-463f-ac47-d8836ec59d90.png)

A React `FunctionalComponent` (`FC`) should always return JSX.  
And other logic should be handled in `useEffect` or other hooks.  
Don't mess with ***render logic*** - this means the return part of the `FC`